src/features/island/buildings/components/building/composters/AdvancedComposter.tsx

```diff
@@ -0,0 +1,132 @@
+import React, { useContext, useState } from "react";
+
+import advancedComposter from "assets/composters/composter_advanced.png";
+import advancedComposterClosed from "assets/composters/composter_advanced_closed.png";
+import { PIXEL_SCALE } from "features/game/lib/constants";
+import { Context } from "features/game/GameProvider";
+import { useActor, useInterpret, useSelector } from "@xstate/react";
+import {
+  CompostingContext,
+  MachineInterpreter,
+  composterMachine,
+  MachineState,
+} from "features/island/buildings/lib/composterMachine";
+import { ComposterModal } from "./ComposterModal";
+import { SUNNYSIDE } from "assets/sunnyside";
+
+const isIdle = (state: MachineState) => state.matches("idle");
+const isComposting = (state: MachineState) => state.matches("composting");
+const isReady = (state: MachineState) => state.matches("ready");
+
+export const AdvancedComposter: React.FC = () => {
+  const { gameService } = useContext(Context);
```

Comment:
You could do the same here with the `gameMachine` and just take the one field you care about which is the building. This would save some re-renders.